common/src/main/java/com/thekillerbunny/goofyplugin/lua/GoofyAPI.java

```diff
@@ -262,6 +263,28 @@ public void loadAvatar(@LuaNotNil String playerUUID) {
         NetworkStuff.getUser(new UserData(uuid));
     }
 
+    @LuaWhitelist
+    @LuaMethodDoc(
+        overloads = {
+            @LuaMethodOverload(
+                argumentTypes = {String.class},
+                argumentNames = {"playerUUID"}
+            )
+        },
+        value = "goofy.get_player_color"
+    )
+    public String getPlayerColor(String playerUUID) {
+        UUID uuid = UUID.fromString(playerUUID);
+        if (AvatarManager.getAvatarForPlayer(uuid) == null) {
+            return "#000000";
+        }
+        Avatar avatar = AvatarManager.getLoadedAvatar(uuid);
```

Comment:
is there a situation where this would return something other than `getAvatarForPlayer`? where did you get this code?

Reply:
That loads the avatar so it works even if someone is offline

Reply:
yes, but why not use its return value rather than fetching it from the local store again?

Reply:
I'll be honest it's me being stupid. Please commit a fix to it to use getLocalAvatar

Reply:
my laptop is entering a state of shock after even thinking of running intellij

Reply:
that'll be another fix PR too